afficher 20 voiture

amostin opened the pull request.

amostin committed:
afficher 20 voiture

amostin commented:
# Premier test de boucle fork
![image](https://user-images.githubusercontent.com/40889985/61187436-59d29900-a671-11e9-817a-a9e1be3d7192.png)

amostin commented:
> # Premier test de boucle fork
> ![image](https://user-images.githubusercontent.com/40889985/61187436-59d29900-a671-11e9-817a-a9e1be3d7192.png)

### Si je comprend bien:
*  j'ai créé 3 processus (car il est marqué 3 fois terminaison du...) 
* et chaque process appel un pere et un fils (les deux lignes contenant les stat voiture)

amostin committed:
affiche 14 voiture

amostin commented:
le fork de fork renvoi 14 voiture si on boucle 3 fois sur create_process et pere fils (3*2+2=14)
![14voiture](https://user-images.githubusercontent.com/40889985/61187620-10377d80-a674-11e9-8acb-585b253abced.PNG)

amostin commented:
J'ai l'impression que on a 3 fois les pere et le fils puis encore une fois les pere et le fils

amostin committed:
affiche 20 voiture

amostin commented:
## Je boucle plus sur le create_process mais sur affiche. Je boucle 10 fois sur affiche dans le fils et 10 dans le pere (10+10=20)
![image](https://user-images.githubusercontent.com/40889985/61187831-f3507980-a676-11e9-85ed-5d79f1144792.png)

amostin commented:
> Mais c'est pas ça qu'on veut... 
👉 Je veux dire que ici on a les 10 premiers tour de la voiture 1 (père) et 10 premiers tour de voiture 2 (fils)

ATHOOS commented:
Je pense qu'on devrait encore pousser un peu plus sur le fork, on a que deux process qui tournent pour l'instant. On devrait essayer de gérer 20 voitures qui affichent chacune leur temps pour 3 tours et qui sont séparées par une ligne (ou un truc du genre).  Comme ca on voit si notre fork fonctionne correctement.

ATHOOS commented:
Et en passant j'ai vite fait géré le cas du num de voiture qui s'incremente et du numero de tour aussi. C'etait juste une question de increment placé la où il faut

ATHOOS commented:
Par contre, un truc que j'ai vu c'est que les tours ne se mettent pas tous dans l'ordre, et encore moins dans l'ordre des pere/enfants. Genre ca mets parfois des temps du fils au milieu du pere sans raison apparente. Je me demande si c'est pas dû au parallelisme des processus

amostin commented:
Tu parles de quel code ? Celui avec un seul fork et 20 lignes ou celui avec 3 fork et 14 lignes ?

amostin commented:
si cest les 3 fork, je pense que cest parce que ya un pere et un fils dans un fils ou un truc du genre non ?

ATHOOS committed:
boucle for de fork

on arrive a avoir 5 voitures qui affichent des temps pour 5 tours

ATHOOS committed:
affichage réglé

ajout d'un sleep afin d'afficher les voitures au fur et a mesure

ATHOOS commented:
x-special/nautilus-clipboard
copy
file:///root/Images/Capture%20d%E2%80%99%C3%A9cran%20de%202019-07-15%2001-23-56.png
affichage réglé

ATHOOS commented:
![Capture d’écran de 2019-07-15 01-23-56](https://user-images.githubusercontent.com/43479148/61190638-9cab6580-a69f-11e9-807a-988d9d980f7f.png)
affichage correct

ATHOOS commented:
![Capture d’écran de 2019-07-15 01-51-56](https://user-images.githubusercontent.com/43479148/61190842-20b31c80-a6a3-11e9-97e8-a2334dd4d388.png)
5 voitures roulent durant 5 avec un affichage correct et avec les "colonnes"

ATHOOS committed:
fork amélioré, code impropre

le code est encore a amélioré, retirer des commentaires, des fonctions inutiles etc

ATHOOS committed:
code clear

suppression des commentaires inutiles, et des fonctions obsolètes

ATHOOS committed:
edit 

correction d'une variable

ATHOOS commented:
Maintenant le fork fonctionne, nos 5 voitures roulent pendant le nombre de tours donné. Ces chiffres sont facilement changeable (ils peuvent même être chacun instancié dans une variable globale, afin de faciliter les changements). Le code a été éclairci, les commentaires inutiles supprimés ainsi que les fonctions devenues inutiles.

ATHOOS commented:
Edit : une erreur persiste, les voitures roulent toujours malgré une sortie de route, je m'en charge tout de suite

ATHOOS commented:
![Capture d’écran de 2019-07-15 02-14-32](https://user-images.githubusercontent.com/43479148/61191074-4c83d180-a6a6-11e9-9fe4-0ab5774c7cf2.png)
Erreur réglée, maintenant un message affiche à quel tour la voiture est sortie de la route et le processus est arrếté

ATHOOS committed:
sortie de route

problème des processus ne s'arretant pas lors d'une sortie de route réglé

amostin commented:
c'est vraiment **NICKEL**
du coup ya moyen de faire ça pour 20 voiture facilement ?

amostin commented:
TEST depuis play (le 5eme process affiche 0 1 ou 2 ligne de voiture
![image](https://user-images.githubusercontent.com/40889985/61191230-1e06f600-a6a8-11e9-8a29-52617204b0be.png)

TEST depuis la console et pas le play (le 5eme affiche le reste bizarrement)
![image](https://user-images.githubusercontent.com/40889985/61191211-c9fc1180-a6a7-11e9-848d-330ebe76bde1.png)

ATHOOS commented:
J'ai aussi l'erreur de temps en temps mais c'est en faisant bcp de test a la suite. Je pense que c'est en rapport avec un sleep qui passe pas correctement. 
J'ai vu aussi une erreur lors du dernier tour, si la voiture se crash au dernier tour, il ne l'affiche pas toujours, je sais pas trop de quoi ca vient par contre.

amostin commented:
# POUR VALIDER LA BRANCHE ON DOIT ARRIVER A CE RESULTAT:
num|s1|s2|s3|tour  |bestour|pit|out|numTour|Tot
---|--|--|--|------|-------|---|---|-------|---
1  |35|41|37|1min53|1min53 |0  |0  |1      |1min53
2  |38|40|38|1min56|1min56 |0  |0  |1      |1min56
3  |38|40|37|1min55|1min55 |0  |0  |1      |1min55
4  |38|35|41|1min54|1min54 |0  |0  |1      |1min54
5  |40|35|39|1min54|1min54 |0  |0  |1      |1min54
... JUSQUA 20

ATHOOS commented:
Edit : rajout d'un sleep afin d'eviter les probleme d'affichage et d'erreur de numero de tour lors d'une sortie de route

ATHOOS committed:
correction erreur sortie de route

rajout d'un sleep afin de corriger les erreurs de sortie de route
